### PR TITLE
Obstacle Detection touches

### PR DESCRIPTION
**Pull request by mhashim6**
some semi-final touches to improve OD module.

### CODE REVIEW

File: obstacledetection/src/main/java/com/caneseeproject/obstacledetection/ObstacleDetectorImpl.kt [hunk 1]
```diff
@@ -12,12 +12,7 @@ class ObstacleDetector : ObstacleDetection {
...
     private val odPortal: SensorPortal = TODO()
 
-    private var cane: Sensor
-
-    /**
-     * Connecting with this cane
-     */
-    fun activate(){
+    override fun activate() {
         cane = odPortal.connect()
```
Comment (yossef-khaled) [inline obstacledetection/src/main/java/com/caneseeproject/obstacledetection/ObstacleDetectorImpl.kt]: where is ### cane initially defined ?

Comment (yossef-khaled) [reply obstacledetection/src/main/java/com/caneseeproject/obstacledetection/ObstacleDetectorImpl.kt]: ok , got it , the factory of ObstacleDetectorImpl class will handle it .

**Comment by mhashim6** [comment]
You’ve been assigned to review this PR, if you’re okay with it you can accept the PR.

**Review by yossef-khaled — APPROVED** [review]
Great , these changes are perfect , i have no objection to them , they can be merged .

**Comment by mhashim6** [comment]
After this, your branch will be merged with the `devel`. Nice work!

**Pull request merged**